cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,328 @@
+        chunksize=hyperparams.chunksize,
+        update_every=hyperparams.update_every,
+        random_state=hyperparams.random_state,
+    )
+    pprint(f"Shows words in each topic: {lda.print_topics(num_words=5)}")
+    file = "lda.model"
+    lda.save(file)
+    return (file,)
+
+
+# %%
+# Word Similarity
+# ===============
+#
+# Word2Vec supports several word similarity tasks out of the box.
+# Here we can print the 10 most similar words to a given word (when running
+# the worflow we will use the word `computer`). We can also easily obtain vectors
+# for terms the model is familiar with”.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def word_similarities(model_ser: FlyteFile[MODELSER_NLP], word: str):
+    model = Word2Vec.load(model_ser.path)
+    wv = model.wv
+    print(f"Word vector for {word}:{wv[word]}")
+    print(f"Most similar words in corpus to {word}: {wv.most_similar(word, topn=10)}")
+
+
+# %%
+# Word Movers Distance
+# ===============
+#
+# WMD enables us to assess the “distance” between two documents in a meaningful way even when
+# they have no words in common. https://radimrehurek.com/gensim/auto_examples/tutorials/run_wmd.html
+# If the sentences have very similar content, the WMD should be low and vice versa.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def word_movers_distance(model_ser: FlyteFile[MODELSER_NLP]) -> float:
+    sentences = [SENTENCE_A, SENTENCE_B]
+    results = []
+    for i in sentences:
+        result = [w for w in utils.tokenize(i) if w not in STOPWORDS]
+        results.append(result)
+    model = Word2Vec.load(model_ser.path)
+    distance = model.wv.wmdistance(*results)
+    print(f"Computing word movers distance for: {SENTENCE_A} and {SENTENCE_B} ")
+    print(f"Word Movers Distance is {distance}")
+    return distance
+
+
+# %%
+# Dimensionality Reduction and Plotting
+# =====================================
+#
+# The word embeddings made by the model can be visualised by reducing dimensionality of
+# the words to 2 dimensions using tSNE. Visualisations can be used to notice semantic
+# and syntactic trends in the data. Since The model used for the visualisation is
+# trained on a small corpus. Thus some of the relations might not be so clear.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def tsne_and_plot(model_ser: FlyteFile[MODELSER_NLP]) -> plotdata:
+    """
+    Trains the word2vec model on the preprocessed corpus with the hyperparameter
+    settings defined in the datclass. The training loss is printed out
+    """
+    model = Word2Vec.load(model_ser.path)
+    num_dimensions = 2
+    vectors = np.asarray(model.wv.vectors)
+    labels = np.asarray(model.wv.index_to_key)
+    tsne = TSNE(n_components=num_dimensions, random_state=0)
+    vectors = tsne.fit_transform(vectors)
+    x_vals = [v[0] for v in vectors]
+    y_vals = [v[1] for v in vectors]
+    plot_with_matplotlib(x_vals, y_vals, labels)
```

Comment:
Could you try using [FlyteDecks](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/flyte_basics/deck.html#sphx-glr-auto-core-flyte-basics-deck-py) in this case? https://blog.flyte.org/orchestration-of-machine-learning-pipelines#heading-10-flyte-decks covers a complex example of FlyteDecks, in case that's helpful.

Reply:
Thanks for that - im having a look